src/Symfony/Bridge/Doctrine/Messenger/DoctrineClearEntityManagerMiddleware.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Messenger;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+
+/**
+ * Clears entity manager after calling all handlers.
+ *
+ * @author Konstanton Myakshin <koc-dp@yandex.ru>
+ *
+ * @experimental in 4.4
+ */
+class DoctrineClearEntityManagerMiddleware implements MiddlewareInterface
+{
+    private $managerRegistry;
+    private $entityManagerName;
+
+    public function __construct(ManagerRegistry $managerRegistry, string $entityManagerName = null)
+    {
+        $this->managerRegistry = $managerRegistry;
+        $this->entityManagerName = $entityManagerName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(Envelope $envelope, StackInterface $stack): Envelope
+    {
+        $entityManager = $this->managerRegistry->getManager($this->entityManagerName);
+
+        if (!$entityManager instanceof EntityManagerInterface) {
+            throw new \InvalidArgumentException(sprintf('The ObjectManager with name "%s" must be an instance of EntityManagerInterface', $this->entityManagerName));
+        }
+
+        try {
+            return $stack->next()->handle($envelope, $stack);
```

Comment:
Imagine the follow scenario:
 - throw an exception in the next handler
 - put entity in that exception
 - have an exception listener
 - try to "fix" the entity or create a record with this entity as relation with cascade persist
 - persist & flush in listener

It's edge-case'y, though it's a scenario I'd image to happen from time to time. What would be the expected and actual result here? As the entity is no longer known to Doctrine, I fear it _can_ create a new one, resulting in oddly duplicated records.

Reply:
I'm not sure that "fixing" entity inside exception listener is a good idea. Usually command handler wrapped in transaction and failed message is requeued for handling later.

For this edge cases you feel free do not use this middleware (it disabled by default).

Reply:
What if the exception was triggered because the entity manager was closed? That would also break sequential handlers.

Reply:
`DoctrinePingConnectionMiddleware` for the rescue https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Doctrine/Messenger/DoctrinePingConnectionMiddleware.php#L56-L58

Reply:
Isn't this effectively breaking the connection of in-memory entities that are still "alive" somewhere?

Reply:
Of course, you should use `EntityManager::refresh()` if some of entities are stored somewhere between messages handling. Or maybe just don't use this middleware for this cases. Note, that this middleware is not enabled by default.